pandas/core/indexes/extension.py

```diff
@@ -110,6 +114,15 @@ def wrapper(self, other):
 
 def make_wrapped_arith_op(opname):
     def method(self, other):
+        if (
+            isinstance(other, Index)
+            and is_object_dtype(other.dtype)
+            and type(other) is not Index
```

Comment:
maybe i am reading this wrong, but you have 2 contraditory conditions here no?

e.g. you likely don't need the 3rd case (actually do you *really* need the 1st case)? e.g. doesn't an object ndarray pass thru here?

Reply:
Does take always copy? I recall an issue about it possibly not in some cases.

Reply:
tricky. it *should* always copy but it doesn't. @jorisvandenbossche was reverting the series.take method deprecation.

Reply:
(sorry, the previous comment about take was on the wrong issue).

---

On this one, we want to be very limited I think. This is written to handle just index **sub**classes with object dtype.

Everything else (including object-dtype Index (not subclasses), ndarray, Series, etc.) should be handled by our usual mechanics.

Reply:
DTI/TDI/PI take calls lib.maybe_indices_to_slices, and if it gets a slice back returns a slice (i.e. view)

Reply:
>On this one, we want to be very limited I think. This is written to handle just index subclasses with object dtype.

wait, we have *Index* subclasses now? I don't think we have *any* testing around this at all.

Reply:
how did this come up?

Reply:
xarray, see #31109.